wolfcrypt/test/test.c

```diff
@@ -49368,6 +49394,49 @@ static int myCryptoDevCb(int devIdArg, wc_CryptoInfo* info, void* ctx)
         info->hmac.hmac->devId = devIdArg;
     }
 #endif
+#if defined(WOLFSSL_CMAC) && !defined(NO_AES)
+    else if (info->algo_type == WC_ALGO_TYPE_CMAC) {
+        if (info->cmac.cmac == NULL) {
+            return NOT_COMPILED_IN;
+        }
+
+        /* set devId to invalid so software is used */
+        info->cmac.cmac->devId = INVALID_DEVID;
+
+        /* Handle one-shot cases */
```

Comment:
Should we add the case of key==null && in!=NULL &out!=null to directly invoke generate_ex?  Not really needed, but could show a decent model for a real cryptocb handler

Reply:
This is true, though I don't like the idea of adding additional stuff into the test cryptoCb handler that isn't actually being tested (as these are just unit tests with no ability to use the key loaded out-of-band). I think it would be best to have a reference implementation of a cryptoCb in documentation that could handle all the cases. 

We also now have more "operations" than our input combinations can support (if we wanted the verify operation to be triggerable as a callback), so we should probably add a bitfield "comand" argument vs deducing operations based on input parameters.

I think this should be saved for a subsequent PR